### PR TITLE
Tabs support for isEmphasized

### LINKED ISSUE #2190: Tabs - support for isEmphasized
# 🙋 Feature Request

Tabs should have support for an `isEmphasized` prop to render an emphasized variant as seen on on [Spectrum's Tabs Documentation](https://spectrum.adobe.com/page/tabs/#Emphasis).

tabs_options_emphasis@2x_1gsgOtyTCwGvtBPTLcym1A_1611635659000

## 🤔 Expected Behavior

From Spectrum:
>By default, tabs are not emphasized. This is optimal for when the tabs component is not the core part of an interface.
> Emphasized tabs have blue text for the selected state for visual prominence and to draw more attention to them. This is optimal for when the selection should call attention, such as the main navigation for a website.

## 💁 Possible Solution

An `isEmphasized` prop should be added to the Tabs component, with a `boolean` type. This should be set to `false` by default if no value is provided.

Styles for emphasized are not currently included in our Tabs styles, so those will need to be added as well.

## 💻 Examples

https://spectrum.adobe.com/page/tabs/#Emphasis

## 🧢 Your Company/Team

RSP

### PR DESCRIPTION
**Pull request by ktabors**
Closes #2190 

Added support for prop `isEmphasized` which is seen in the code as the class `.spectrum-Tabs--emphasized` to Tabs in quiet, hover, compact, and vertical. `isEmphasized` is not supported when collapsed because we are using a picker which does not support `isEmphasized`, but the selection indicator under the picker does show `isEmphasized`.

I primarily wanted to add stories in Chromatic Storybook. Chromatic didn't have icons stories or collapse, so those and a few basic ones are in the non-Chromatic Storybook.

No tests were added because other equivalent features didn't have them.

Related Docs PR #3125

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Run chromatic, there are new stories.
Check Storybook for the new `isEmphasized` stories. Please test them.
You could also checkout the related Docs PR #3125

## 🧢 Your Project:
RSP

### CODE REVIEW

File: packages/@adobe/spectrum-css-temp/components/tabs/skin.css [hunk 1]
```diff
@@ -82,6 +114,10 @@ governing permissions and limitations under the License.
 
   .spectrum-Tabs-selectionIndicator {
     background-color: var(--spectrum-tabs-quiet-selection-indicator-color);
+
+    &.spectrum-Tabs--emphasized {
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/tabs/skin.css:118]: i don't think this is the selector you want, it'll generate something like
`.spectrum-Tabs .spectrum-Tabs-selectionIndicator.spectrum-Tabs--emphasized`
double hyphen means state, single means hierarchy/namespace
I'd expect this to be
`.spectrum-Tabs.spectrum-Tabs--emphasized .spectrum-Tabs-selectionIndicator`
you may need to re-arrange some of this to accomplish that, ampersand just means, take the selector (computed) above and insert it in place of the ampersand

Comment (ktabors) [reply packages/@adobe/spectrum-css-temp/components/tabs/skin.css]: thanks! refactored

File: packages/@adobe/spectrum-css-temp/components/tabs/index.css [hunk 2]
```diff
@@ -15,6 +15,14 @@ governing permissions and limitations under the License.
 :root {
   --spectrum-tabs-item-height: calc(var(--spectrum-tabs-height) - var(--spectrum-tabs-rule-height));
   --spectrum-tabs-compact-item-height: calc(var(--spectrum-tabs-quiet-compact-height) - var(--spectrum-tabs-rule-height));
+  --spectrum-tabs-emphasized-text-color: var(--spectrum-global-color-blue-600);
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/tabs/index.css:18]: we can put these into `:root` in the skin.css file I believe

File: packages/@react-spectrum/tabs/stories/Tabs.stories.tsx [hunk 3]
```diff
@@ -73,6 +73,18 @@ storiesOf('Tabs', module)
...
   )
+  .add(
+    'isEmphasized: true',
+    () => render({isEmphasized: true})
+  )
+  .add(
+    'isEmphasized: true, icons, isQuiet: true',
+    () => renderWithIcons({isEmphasized: true, isQuiet: true})
+  )
+  .add(
```
Comment (reidbarber) [inline packages/@react-spectrum/tabs/stories/Tabs.stories.tsx:84]: Might be good to throw in some with `desnsity: compact`

Comment (ktabors) [reply packages/@react-spectrum/tabs/stories/Tabs.stories.tsx:84]: Added them to chromatic stories.

**Review by snowystinger — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]

**Pull request merged**